### PR TITLE
Add ability to change Discovery Prefix via WebUI or via mosquitto_pub

### PR DESCRIPTION
**Pull request by puterboy**
## Description:
Changes made as discussed in https://github.com/1technophile/OpenMQTTGateway/issues/1996
Specifically:
1. Declaration `discovery_Topic` deprecated (but still working) and changed to `discovery_Prefix` to avoid confusion with `discovery_topic`
2. New *variable* `discovery_prefix` introduced that defaults on new installs to discovery_Prefix (or discovery_Topic if present), is stored in NVS, and can be changed via WebUI  (under `Configure MQTT`) or via mosquitto_pub, using something like:
```
mosquitto_pub -t "home/<GATEWAY_NAME>/commands/MQTTtoSYS/config" -m '{  "discovery_prefix": "<NEW_DISCOVERY_PREFIX" }'
```

I tested it using an esp32 with `esp32dev-ble` and the functionality works as described.
Let me know if you are ok with the PR.

## Checklist:
  - [X ] The pull request is done against the latest development branch
  - [X ] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: main/main.ino [hunk 1]
```diff
@@ -1332,6 +1332,7 @@ void setup() {
   Log.trace(F("OpenMQTTGateway ip: %s" CR), WiFi.localIP().toString().c_str());
   Log.trace(F("OpenMQTTGateway index %d" CR), cnt_index);
   Log.trace(F("OpenMQTTGateway mqtt topic: %s" CR), mqtt_topic);
+  Log.trace(F("OpenMQTTGateway mqtt topic: %s" CR), discovery_prefix);
```
Comment (1technophile) [inline main/main.ino]: OpenMQTTGateway mqtt discovery prefix

**Comment by 1technophile** [comment]
Thanks, could you update the documentation also.

The build ko for the AWS environment is because some environments do not use discovery. Your change needs to be included when the macro ZmqttDiscovery is defined.

Finally check the box for the DCO in the PR and we will be good

**Comment by puterboy** [comment]
- Corrected the typo where mqtt_topic was used instead of discovery_prefix
- Added #ifdef ZmqttDiscovery in all *new* places that reference discovery_prefix (note if a line of code previously referenced discovery_Topic and was not conditioned before then I left it as is now)
- Added discovery_prefix key to the mosquito_pub documentation (Note the documentation doesn't address setting "MQTT Base Topic" except in initial startup where "Discovery Prefix" is not relevant -- so I didn't know where or if I should document "Discovery Prefix" for the regular WebUI

**Comment by puterboy** [comment]
Not sure what the lint formatting issues are - I tried to copy how you spaced and indented macro if/else/end statements.

**Comment by 1technophile** [comment]
So as to format automatically your document you have to add the "clang-Format" extension to VSCode, once done, you can format the code by doing a right click into the code file window and clicking "Format document".

**Comment by puterboy** [comment]
I was actually doing the editing in the web version of github – not VS.

Do you have the ability to fix the formatting on my patch?

**Comment by 1technophile** [comment]
Yes sure

**Comment by puterboy** [comment]
Thanks!

**Pull request merged**